src/pages/Dashboard/components/DashboardHeader/WalletBalance.tsx

```diff
@@ -0,0 +1,62 @@
+import { Flex, Skeleton } from '@chakra-ui/react'
+import type { ResponsiveValue } from '@chakra-ui/system'
+import { bnOrZero } from '@shapeshiftoss/chain-adapters'
+import type { Property } from 'csstype'
+import { memo, useMemo } from 'react'
+import { Amount } from 'components/Amount/Amount'
+import { useFetchOpportunities } from 'components/StakingVaults/hooks/useFetchOpporunities'
+import { Text } from 'components/Text'
+import {
+  selectClaimableRewards,
+  selectEarnBalancesUserCurrencyAmountFull,
+  selectPortfolioLoading,
+  selectPortfolioTotalUserCurrencyBalanceExcludeEarnDupes,
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const balanceFontSize = { base: '2xl', md: '4xl' }
+const balanceFlexDir: ResponsiveValue<Property.FlexDirection> = {
+  base: 'column-reverse',
+  md: 'column',
+}
+const portfolioTextAlignment: ResponsiveValue<Property.AlignItems> = {
+  base: 'center',
+  md: 'flex-start',
+}
+export const WalletBalance = memo(() => {
+  const { isLoading: isOpportunitiesLoading } = useFetchOpportunities()
```

Comment:
as this is called in multiple places we need to verify that we aren't refetching the same data repeatedly, especially since its the biggest performance bottleneck of the whole app

Reply:
@woodenfurniture doesn't look like this should be the responsibility of consumers? i.e https://github.com/shapeshift/web/pull/6409

Reply:
Just read your review comment at the end of the PR, jinxed